src/ResolvedApi.ts

```diff
@@ -0,0 +1,236 @@
+
+export const PREVIEW_COOKIE = 'io.prismic.preview';
+export const EXPERIMENT_COOKIE = 'io.prismic.experiment';
+
+export interface Ref {
+  ref: string;
+  label: string;
+  isMasterRef: string;
+  scheduledAt: string;
+  id: string;
+}
+
+export interface ApiData {
+  refs: Ref[];
+  bookmarks: { [key: string]: string };
+  types: { [key: string]: string };
+  tags: string[];
+  forms: { [key: string]: Form };
+  experiments: any;
+  oauth_initiate: string;
+  oauth_token: string;
+  version: string;
+  licence: string;
+}
+
+export interface PreviewResponse {
+  mainDocument?: string;
+}
+
+export interface QueryOptions {
+  [key: string]: string | number;
+}
+
+export interface ResolvedApiOptions {
+  req?: any;
+}
+
+export default class ResolvedApi implements Client {
+  data: ApiData;
+  masterRef: Ref;
+  experiments: Experiments;
+  options: ResolvedApiOptions;
+  httpClient: HttpClient;
+  bookmarks: { [key: string]: string };
+  refs: Ref[];
+  tags: string[];
+  types: { [key: string]: string };
+
+  constructor(data: ApiData, httpClient: HttpClient, options: ResolvedApiOptions) {
+    this.data = data;
+    this.masterRef = data.refs.filter(ref => ref.isMasterRef)[0];
+    this.experiments = new Experiments(data.experiments);
+    this.bookmarks = data.bookmarks;
+    this.httpClient = httpClient;
+    this.options = options;
+    this.refs = data.refs;
+    this.tags = data.tags;
+    this.types = data.types;
+  }
+
+  /**
+   * Returns a useable form from its id, as described in the RESTful description of the API.
+   * For instance: api.form("everything") works on every repository (as "everything" exists by default)
+   * You can then chain the calls: api.form("everything").query('[[:d = at(document.id, "UkL0gMuvzYUANCpf")]]').ref(ref).submit()
+   */
+  form(formId: string): SearchForm | null {
+    const form = this.data.forms[formId];
+    if (form) {
+      return new SearchForm(form, this.httpClient);
+    }
+    return null;
+  }
+
+  everything(): SearchForm {
+    const f = this.form('everything');
+    if (!f) throw new Error('Missing everything form');
+    return f;
+  }
+
+  /**
+   * The ID of the master ref on this prismic.io API.
+   * Do not use like this: searchForm.ref(api.master()).
+   * Instead, set your ref once in a variable, and call it when you need it; this will allow to change the ref you're viewing easily for your entire page.
+   */
+  master(): string {
+    return this.masterRef.ref;
+  }
+
+  /**
+   * Returns the ref ID for a given ref's label.
+   * Do not use like this: searchForm.ref(api.ref("Future release label")).
+   * Instead, set your ref once in a variable, and call it when you need it; this will allow to change the ref you're viewing easily for your entire page.
+   */
+  ref(label: string): string | null {
+    const ref = this.data.refs.filter(ref => ref.label === label)[0];
+    return ref ? ref.ref : null;
+  }
+
+  currentExperiment(): Experiment | undefined {
```

Comment:
why undefined here instead of null like everywhere else?

Reply:
We could use `null` instead of `undefined`